wallet/src/main/scala/org/bitcoins/wallet/api/NeutrinoWalletApi.scala

```diff
@@ -0,0 +1,99 @@
+package org.bitcoins.wallet.api
+
+import org.bitcoins.core.gcs.GolombFilter
+import org.bitcoins.core.protocol.BlockStamp
+import org.bitcoins.core.protocol.blockchain.Block
+import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.crypto.{DoubleSha256Digest, DoubleSha256DigestBE}
+import org.bitcoins.wallet.api.NeutrinoWalletApi.BlockMatchingResponse
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait NeutrinoWalletApi extends WalletApi {
```

Comment:
So should this really `extends WalletApi` or should it require a `WalletApi` to be injected into this trait with [scala's self type annotation](https://www.baeldung.com/scala/self-type-annotation), i.e. 

```scala
trait NeutrinoWalletApi { self : WalletApi => ... } 
```

As I think we have discussed online, it doesn't seem like the neutrino APIs are directly related to the wallet, rather they are a synchronization mechanism for _any_ wallet type.